src/request.ts

```diff
@@ -123,12 +130,22 @@ class Request {
           }
           return this.req(options);
         })
-        .then((res: AxiosResponse<TResponse>): void => {
-          if (resolve) {
-            resolve(res.data);
+        .then((res: any): void => {
+          if (isLargeFile) {
```

Comment:
How about we made it not specific to large files, but instead check the requested `responseType` and if it's been set to `stream`, we resolve the promise with raw data? That way we can use it in different context rather than making it specific to large files, which is just one use case for which we need this.